chain/builder.go

```diff
@@ -59,61 +77,53 @@ func HandlePreExecute(log logging.Logger, err error) bool {
 	}
 }
 
-// TODO: This code is terrible and will be removed during the Vryx integration.
-func BuildBlock(
-	ctx context.Context,
-	vm VM,
-	parent *StatefulBlock,
-) (*StatefulBlock, error) {
-	ctx, span := vm.Tracer().Start(ctx, "chain.BuildBlock")
-	defer span.End()
-	log := vm.Logger()
+type buildBlock struct {
+	parentID  ids.ID
+	timestamp int64
+	height    uint64
+	txs       []*Transaction
+	stateRoot ids.ID
+}
```

Comment:
add an intermediate type rather than creating a partially initialized instance of a block

Reply:
Will change to local variables